src/index.js

```diff
@@ -357,10 +361,14 @@ router
   })
   .get("/likes/:feed", async ctx => {
     const { feed } = ctx.params;
-
     const likes = async ({ feed }) => {
-      const messages = await post.likes({ feed });
-      return listView({ messages });
+      const pendingMessages = post.likes({ feed });
+      const pendingName = about.name(feed);
+      return likesView({
+        messages: await pendingMessages,
+        feed,
+        name: await pendingName
+      });
```

Comment:
This is a way to run promises in parallel without using Promise.all

Reply:
🤯

Reply:
@cinnamon-bun I just tried this in another project and it doesn't seem like it's doing the `Promise.all()` thing. Am I doing it wrong?

```javascript
const crypto = require("crypto");

const bullshit = () =>
  new Promise(resolve => {
    const id = crypto.randomBytes(8).toString("hex");
    console.log(`start ${id}`);
    const timeout = (Math.random() + 1) * 1000; // 1-2 seconds
    setTimeout(() => {
      console.log(`end ${id}`);
      resolve();
    }, timeout);
  });

const main = async () => {
  const results = {
    x: await bullshit(), // takes 1-2 seconds
    y: await bullshit(), // takes 2-4 seconds
    z: await bullshit()  // takes 3-6 seconds
  };
};

main();
```

Reply:
It works by separating things into two phases:
1. generate some promises so they start doing work in the background, but don't await them yet
2. await the promises

Like:
```js
const main = async () => {
  // PHASE 1
  // start them all in parallel but don't wait for them to finish yet
  const bs1promise = bullshit();
  const bs2promise = bullshit();
  const bs3promise = bullshit();

  // PHASE 2
  // this large expression will block until all the promises inside it are ready
  const results = {
    x: await bs1promise,
    y: await bs2promise,
    z: await bs3promise
  };
};
```

In your example I think the laungage doesn't take advantage of the obvious lack of dependencies between x, y, and z, it just scans linearly through the syntax, so we get stuck `await`ing the first one and it hasn't started the rest yet.

Reply:
Oh, that makes sense -- thanks for explaining that to me! It's very clever.